src/features/midi/index.tsx

```diff
@@ -142,30 +177,45 @@ class MidiState {
   }
 
   handleKeyDown(e: KeyboardEvent) {
+    if (e.metaKey || e.ctrlKey || e.altKey) {
+      return
+    }
+    const key = e.key.toLowerCase()
+
+    if (key in qwertyKeyConfig) {
+      e.preventDefault()
+      e.stopPropagation()
+    }
+
     // Some OSes / browsers will automatically repeat a letter when held down.
     // We don't want to count those.
-    if (e.repeat || e.metaKey || e.ctrlKey || e.altKey) {
+    if (e.repeat) {
       return
     }
 
-    // TODO: what if octave switch while note is held down
-    // must release all currently pressed notes.
-    const key = e.key.toLowerCase()
     if (key === 'arrowup') {
-      this.octave = Math.min(7, this.octave + 1)
+      this.octaveDiff = Math.min(2, this.octaveDiff + 2)
```

Comment:
Can you explain why you prefer middle c + diff, versus the single piece of state (octave). Also doesn't this `2` max mean octave 7 can't be reached? Same question for the min being `-2`, means octave 1 can't be reached.